M2/Macaulay2/m2/matrix.m2

```diff
@@ -119,6 +119,7 @@ Matrix == Matrix := (f,g) -> (
 Matrix == Number :=
 Matrix == RingElement := (m,f) -> m - f == 0		    -- slow!
 Matrix == ZZ := (m,i) -> if i === 0 then rawIsZero m.RawMatrix else m - i == 0
+Vector == ZZ := (v,i) -> matrix v == i
```

Comment:
1. Did you mean to also add `ZZ == Vector`? This might be a simple way to do both:
```suggestion
Vector == ZZ := ZZ == Vector := (u, v) -> matrix u == matrix v
```
2. What you wrote also makes `vector {1} == 1` to be true, which is probably fine, but just making sure this is intended.
3. Is this automatically caught in a generic documentation node?

Reply:
1.  I didn't think to, but that's a good idea!  I just pushed a new version adding this, and also supporting comparison using `==` with `Number` and `RingElement`, matching the corresponding `Matrix` methods.
2.  Yes, that's intended.
3. No, but I think that's a wider issue with all the `==` methods.  None of them seem to be documented!

Reply:
While you're at it, maybe also define `Vector == Matrix`?

I don't particularly care about all of these nodes being documented, as long as it's obvious what they do. I do care about them not "missing documentation" though, so either marking as undocumented or globbing all missing nodes involving `==` seem like fine solutions to me. This would still allow documenting the ones that are noteworthy, like for modules/ideals/sheaves.

Reply:
Yeah, `==` is really bad for missing documentation nodes!  https://macaulay2.com/doc/Macaulay2/share/doc/Macaulay2/Macaulay2Doc/html/__eq_eq.html

Some sort of automatic way for adding otherwise undocumented methods for common operators like `+`, `-`, and `==` to the main node for that operator sounds like great idea.  But I think it's outside the scope of this PR.

Reply:
Okay, then maybe don't touch the documentation at all.

Reply:
I figured I'd at least document the `+` and `-` methods that I added.  The spacing between all the various keys for these nodes were super messy and inconsistent, so I decided I might as well clean it up a bit.

Reply:
If you want to clean up the messy documentation nodes then use globbing, otherwise why touch a hundred lines just to fix the indentation?

Reply:
Fair enough.  I removed that commit and just added the new keys for now.